eventsourcing/src/main/java/org/axonframework/eventsourcing/SnapshotterSpanFactory.java

```diff
@@ -0,0 +1,31 @@
+package org.axonframework.eventsourcing;
+
+import org.axonframework.tracing.Span;
+
+/**
+ * Span factory that creates spans for the {@link Snapshotter}.
+ * You can customize the spans of the snapshotter by creating your own implementation.
+ *
+ * @author Mitchell Herrijgers
+ * @since 4.9.0
+ */
+public interface SnapshotterSpanFactory {
+    /**
+     * Creates a new {@link Span} which represents the scheduling of snapshot creation to the snapshotter's executor.
+     *
+     * @param aggregateType       The aggregate's type.
+     * @param aggregateIdentifier The aggregate's identifier.
+     * @return A {@link Span} representing the scheduling of snapshot creation.
+     */
+    Span createScheduleSnapshotSpan(String aggregateType, String aggregateIdentifier);
+
+    /**
+     * Creates a new {@link Span} which represents the actual creation of a snapshot. The creation
+     * of a snapshot might be done in a separate thread depending on the configuration of the  {@link AbstractSnapshotter}.
```

Comment:
Why refer to the `AbstractSnapshotter` i.o. the `Snapshotter` interface? Because only the former conclusively uses this span factory?

Reply:
I did this because this has configuration for the execution model. However, I can also refer to the 'implementation of the Snapshotter', so adjusting